### PR TITLE
[WIP] refactor(button): surf-1889

### PR DESCRIPTION
**Pull request by lalithkarikelli**
JIRA:SURF-1889

### LGTM's
* [ ] Dev LGTM
* [ ] Dev LGTM

### CODE REVIEW

File: src/scss/components/button/_index.scss [hunk 1]
```diff
@@ -1 +1,315 @@
...
+  @mixin medium() {
+    font-size: 0.875rem;
+    padding: 8px 12px;
+  }
+
+  @mixin large() {
+    font-size: 1rem;
+    padding: 1rem;
+  }
+}
```
Comment (CITguy) [inline src/scss/components/button/_index.scss]: Remove lines 3-142.

Button mixins have already been defined in SCSS.

File: src/scss/components/button/_index.scss [hunk 2]
```diff
@@ -1 +1,315 @@
...
+    border-radius: 0;
+
+    + .hxBtn {
+      border-left: 0;
+    }
+  }
+
+  &.hxPrimary {
+    > .hxBtn {
+      @extend .hxBtn.hxPrimary;
```
Comment (CITguy) [inline src/scss/components/button/_index.scss]: This looks correct, but we'll need to double-check that this is extending in SCSS in the same way it extends in LESS.

The same goes for each `@extend` below, too (compare the CSS in `dist/styles/helix-ui.css` with that of https://unpkg.com/helix-ui@0.18.0-rc.0/dist/styles/helix-ui.css).

File: src/scss/components/button/_index.scss [hunk 3]
```diff
@@ -1 +1,315 @@
...
+    &:hover, &:active  {
+      background-color: transparent;
+
+      hx-icon {
+        color: $gray-700;
+      }
+    }
+
+    &:focus {
+      box-shadow: 0 0 4px 0 fade($cyan-500, 50%);
```
Comment (CITguy) [inline src/scss/components/button/_index.scss]: `fade()` needs to be replaced with the SCSS function `rgba()`

```scss
box-shadow: 0 0 4px 0 rgba($cyan-500, 0.5);
```

File: src/scss/components/button/_index.scss [hunk 4]
```diff
@@ -1 +1,315 @@
...
+
+  }
+  &:active {
+    @extend .hxBtn:active;
+
+  }
+  &:focus {
+    @extend .hxBtn:focus;
+
+    box-shadow: 0 0 4px 2px fade($cyan-500, 60%);
```
Comment (CITguy) [inline src/scss/components/button/_index.scss]: `fade()` needs replaced and updated with `rgba()`

```scss
box-shadow: 0 0 4px 2px rgba($cyan-500, 0.6);
```

File: src/scss/components/button/_index.scss [hunk 5]
```diff
@@ -1 +1,315 @@
-// TBD... button styles
+@import "config";
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/button/_index.scss]: Screenshot from 2019-10-31 18-28-06
----------------------------------------------------------------------
am not facing any lint issues. PFA for reference

File: src/scss/components/button/_index.scss (hunk 1 above)
Comment (lalithkarikelli) [reply src/scss/components/button/_index.scss]: Changes have been done, please review.

File: src/scss/components/button/_index.scss (hunk 3 above)
Comment (lalithkarikelli) [reply src/scss/components/button/_index.scss]: Changes have been done, please review.

File: src/scss/components/button/_index.scss (hunk 4 above)
Comment (lalithkarikelli) [reply src/scss/components/button/_index.scss]: Changes have been done, please review.

**Comment by CITguy** [comment]
Marked "DO NOT MERGE" because scope of work has grown beyond original plan. We'll be handing over work for this story to a senior developer.

**Comment by CITguy** [comment]
Closing PR. See #659

**Pull request closed without merging**